### PR TITLE
MGDAPI-4457 Migrate from Ginkgo v1 to v2

### PR DESCRIPTION
**Pull request by Fiona-Waters**
## Overview

Jira: [MGDAPI-4457](https://issues.redhat.com/browse/MGDAPI-4457)
 - Migrate ginkgo to v2
We are currently using Ginkgo v1.16.5 but the latest version is v2.4.1 and [v2](https://github.com/onsi/ginkgo/releases/tag/v2.0.0) of Ginkgo has been supported since December 30, 2021. Since v1.x is deprecated, we should move to the latest version of Ginkgo.
- 2 changes were also made to README.md

## Verification
 - Ensuring that this PR is passing the automated tests along with reviewing the changed files should be sufficient verification.
- Alternatively the changes can also be verified by provisioning a cluster, installing CRO via OLM using this [index image](http://quay.io/rh_ee_fwaters/cloud-resource-operator:v0.40.0), and then running the tests against it.

### CODE REVIEW

**Review by KevFan — COMMENTED** [review]
Looks good, just have one minor comment :+1:

File: README.md [hunk 1]
```diff
@@ -35,6 +35,7 @@ Prerequisites:
 - [operator-sdk](https://github.com/operator-framework/operator-sdk) version v1.14.0.
 - [git-secrets](https://github.com/awslabs/git-secrets) - for preventing cloud-provider credentials being included in 
 commits
+- [OPM](https://docs.openshift.com/container-platform/4.6/cli_reference/opm-cli.html)
```
[the remark this answers is not in the document]
Comment (Fiona-Waters) [reply README.md]: Yes definitely, thank you for catching that. I have updated it.

**Comment by KevFan** [comment]
Changes looks good and prow checks has passed :+1:  Nice work ! :+1: 

/lgtm

**Pull request merged**